src/docker-utils.coffee

```diff
@@ -38,7 +38,7 @@ applyDelta = (imgSrc, deltaUrl, { requestTimeout, applyTimeout, resumeOpts }, on
 				deltaStream = dockerDelta.applyDelta(imgSrc, timeout: applyTimeout)
 				res.pipe(deltaStream)
 				.on('id', resolve)
-				.on('error', reject)
+				.on('error', req.destroy.bind(req))
```

Comment:
How does the Promise get rejected in this case? Does `req.destroy` make `req` fire an error event?

Reply:
Yeah, `req.destroy` will be called with an `Error` which will make `ResumableRequest` emit an error event, which will eventually reject the promise. That's the clearest way I could think of that allowed both `ResumableRequest` and the delta stream to be properly closed (and also kill rsync if needed).